pkg/quota/v1/evaluator/core/persistent_volume_claims.go

```diff
@@ -159,23 +159,46 @@ func (p *pvcEvaluator) Usage(item runtime.Object) (corev1.ResourceList, error) {
 		result[storageClassClaim] = *(resource.NewQuantity(1, resource.DecimalSI))
 	}
 
-	// charge for storage
-	if request, found := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; found {
-		roundedRequest := request.DeepCopy()
+	requestedStorage := p.getStorageUsage(pvc)
+	if requestedStorage != nil {
+		result[corev1.ResourceRequestsStorage] = *requestedStorage
+		// charge usage to the storage class (if present)
+		if len(storageClassRef) > 0 {
+			storageClassStorage := corev1.ResourceName(storageClassRef + storageClassSuffix + string(corev1.ResourceRequestsStorage))
+			result[storageClassStorage] = *requestedStorage
+		}
+	}
+
+	return result, nil
+}
+
+func (p *pvcEvaluator) getStorageUsage(pvc *corev1.PersistentVolumeClaim) *resource.Quantity {
+	var result *resource.Quantity
+	roundUpFunc := func(i *resource.Quantity) *resource.Quantity {
+		roundedRequest := i.DeepCopy()
 		if !roundedRequest.RoundUp(0) {
 			// Ensure storage requests are counted as whole byte values, to pass resourcequota validation.
 			// See http://issue.k8s.io/94313
-			request = roundedRequest
+			return &roundedRequest
 		}
+		return i
+	}
 
-		result[corev1.ResourceRequestsStorage] = request
-		// charge usage to the storage class (if present)
-		if len(storageClassRef) > 0 {
-			storageClassStorage := corev1.ResourceName(storageClassRef + storageClassSuffix + string(corev1.ResourceRequestsStorage))
-			result[storageClassStorage] = request
+	if userRequest, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; ok {
+		result = roundUpFunc(&userRequest)
+	}
+
+	if len(pvc.Status.AllocatedResources) == 0 {
+		return result
+	}
+
+	// if AllocatedResources is set and is greater than user request, we should use it.
+	if allocatedRequest, ok := pvc.Status.AllocatedResources[corev1.ResourceStorage]; ok {
+		if allocatedRequest.Cmp(*result) > 0 {
```

Comment:
could the result be nil here? Since we do the check few lines above.

Reply:
you mean `result` variable? You can't create a PVC without specifying setting `resources.requests` field, so I do not think it can be nil.

Reply:
ok makes sense, just asking because I saw this check:

```go
if userRequest, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; ok {
	result = roundUpFunc(&userRequest)
}
```